### PR TITLE
[RECONSTRUCTION] [CLANG] Fix warnings reported by llvm16 in CLANG IBs

### PR DESCRIPTION
**Pull request by aandvalenzuela**
Fixes set but unused warnings from clang-16 in CLANG IBs.

### CODE REVIEW

File: TrackPropagation/RungeKutta/src/RKAdaptiveSolver.icc [hunk 1]
```diff
@@ -40,11 +39,8 @@ typename RKAdaptiveSolver<T, StepWithPrec, N>::Vector RKAdaptiveSolver<T, StepWi
     float acc = tryStep.second;
     //assert(eps>0);
     // assert(acc>=0);
-    nsteps++;
     if (acc < eps || std::abs(stepSize) < std::abs(remainigStep) * 0.1f) {
       if (std::abs(remainigStep - stepSize) < 0.5f * eps) {
-        //	if (verbose()) std::cout << "Accuracy reached, and full step taken in "
```
Comment (VinInn) [inline TrackPropagation/RungeKutta/src/RKAdaptiveSolver.icc:46]: maybe some form of ```#ifdef``` would be more appropriate than just removing those debug statements

Comment (aandvalenzuela) [reply TrackPropagation/RungeKutta/src/RKAdaptiveSolver.icc:46]: Sure, I already applied your suggestion. Thanks @VinInn!

**Comment by mandrenguyen** [comment]
+reconstruction

**Comment by rappoccio** [comment]
+1

**Pull request merged**